protocol/contracts/beanstalk/sun/SeasonFacet/Weather.sol

```diff
@@ -168,18 +179,125 @@ contract Weather is Sun {
+            floodPodline();
+
             if (s.r.roots > 0) {
-                // initalize sopWell if it is not already set.
-                if (s.sopWell == address(0)) s.sopWell = well;
-                sop();
+                (
+                    WellDeltaB[] memory wellDeltaBs,
+                    uint256 totalPositiveDeltaB,
+                    uint256 totalNegativeDeltaB,
+                    uint256 positiveDeltaBCount
+                ) = getWellsByDeltaB();
+                wellDeltaBs = calculateSopPerWell(
+                    wellDeltaBs,
+                    totalPositiveDeltaB,
+                    totalNegativeDeltaB,
+                    positiveDeltaBCount
+                );
+
+                for (uint i; i < wellDeltaBs.length; i++) {
+                    sopWell(wellDeltaBs[i]);
+                }
+            }
+        }
+    }
+
+    /**
+     * @notice Floods the field, up to 0.1% of the total Bean supply worth of pods.
+     */
+    function floodPodline() private {
+        // Make 0.1% of the total bean supply worth of pods harvestable.
+
+        uint256 totalBeanSupply = C.bean().totalSupply();
+        uint256 sopFieldBeans = totalBeanSupply.div(FLOOD_PODLINE_PERCENT_DENOMINATOR); // 1/1000 = 0.1% of total supply
+
+        // Note there may be cases where zero harvestable pods are available. For clarity, the code will still emit an event
+        // but with zero sop field beans.
+        uint256 maxHarvestable = s.f.pods.sub(s.f.harvestable);
+        sopFieldBeans = sopFieldBeans > maxHarvestable ? maxHarvestable : sopFieldBeans;
+
+        s.f.harvestable = s.f.harvestable.add(sopFieldBeans);
+        C.bean().mint(address(this), sopFieldBeans);
+
+        emit SeasonOfPlentyField(sopFieldBeans);
+    }
+
+    function getWellsByDeltaB()
+        public
+        view
+        returns (
+            WellDeltaB[] memory wellDeltaBs,
+            uint256 totalPositiveDeltaB,
+            uint256 totalNegativeDeltaB,
+            uint256 positiveDeltaBCount
+        )
+    {
+        address[] memory wells = LibWhitelistedTokens.getCurrentlySoppableWellLpTokens();
+        wellDeltaBs = new WellDeltaB[](wells.length);
+
+        for (uint i = 0; i < wells.length; i++) {
+            wellDeltaBs[i] = WellDeltaB(wells[i], LibWellMinting.currentDeltaB(wells[i]));
+            if (wellDeltaBs[i].deltaB > 0) {
+                totalPositiveDeltaB += uint256(wellDeltaBs[i].deltaB);
+                positiveDeltaBCount++;
+            } else {
+                totalNegativeDeltaB += uint256(-wellDeltaBs[i].deltaB);
+            }
+        }
+
+        // Sort the wellDeltaBs array
+        quickSort(wellDeltaBs, 0, int(wellDeltaBs.length - 1));
+    }
+
+    // Reviewer note: This works, but there's got to be a way to make this more gas efficient
+    function quickSort(
+        WellDeltaB[] memory arr,
+        int left,
+        int right
+    ) public pure returns (WellDeltaB[] memory) {
```

Comment:
i'd honestly make this a private, given it doesn't seem as useful to expose + save bytecode size

Reply:
Updated: https://github.com/BeanstalkFarms/Beanstalk/pull/870/commits/71da3435e23131a913fc5da74e88e1c49bfc913e